### PR TITLE
Document all of the schemas in their own README

### PR DESCRIPTION
**Pull request by Athanasius**
To aid potential senders (mainly) and listeners (less specifically so, but they should benefit from the sender documentation) let's document all of the schemas individually so it's clear what must be:

- Elided
- Added
- Changedd

from the data source(s).

### CODE REVIEW

File: schemas/blackmarket-README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+**This schema is deprecated.  The same data has been made available via the
+`prohibited` array in the [commodity schema](./commodity-v3.0.json) since 
+September 2017.**
+
+What follows below is only for historical curiosity.
+
+---
+
+Here we document how to take data from an ED `BlackMArket` Journal Event and
```
Comment (Tkael) [inline schemas/blackmarket-README.md:13]: `BlackMArket` should be `BlackMarket`. ;-)

Comment (Tkael) [reply schemas/blackmarket-README.md:13]: There is no `Blackmarket` journal event. Per line 20 below, "the primary data source for this schema is the ED Journal event `MarketSell`."

File: schemas/blackmarket-README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+have the key renamed to `prohibited`.
+
+#### marketID
+The Journal documentation says this is `MarketID`, but in the schema the 
+`m` is lower case.
+
+### Elisions
+You MUST remove the following key/value pairs from the data:
+
+  - `Count`
```
Comment (Tkael) [inline schemas/blackmarket-README.md:38]: `Count` may be potentially useful in some cases. Historically, tracking UA bombing for example.

File: schemas/commodity-README.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+- Skip commodities with `"categoryname": "NonMarketable"` (i.e.
+  Limpets - not purchasable in station market) or a *non-empty*`"legality":` 
+  string (not normally traded at this station market).
+
+#### Item Category
+Remove not only the `Category_Localised` key/value, as above, but also the
+`Category` key/value pair from each Item.
+
+### Augmentations
```
Comment (Tkael) [inline schemas/commodity-README.md:51]: `statusFlags` is an optional augmentation that can be used to identify `Producer`, `Consumer`, and `Rare` goods.

File: schemas/commodity-README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+   lagging behind the game client state and thus should not be used.
+
+Note that CAPI `/market` data will sometimes have the `StatusFlasg` per 
+item, which are defined as optional in this schema (because they're not in 
+the Market.json data).  You SHOULD include this data in your message if 
+using CAPI as the source.
+
+Now you will need to construct the necessary additional fields:
+
+#### CAPI horizons flag
```
Comment (Tkael) [inline schemas/commodity-README.md:78]: If your app only refreshes CAPI in response to active journal events and does not refresh CAPI `/market` data while the player is not actively playing the game then you may use the value from the Journal `LoadGame` event.

File: schemas/commodity-README.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+You will need to check if any of the economies from the `/market` data have 
+a `name` are `Colony`, if so, set this flag true.
+
+Additionally, you should retrieve the CAPI `/shipyard` endpoint and check if 
+any of the listed ships or modules have a `sku` value of 
+`ELITE_HORIZONS_V_PLANETARY_LANDINGS`.  If so, set this flag true.
+
+#### CAPI odyssey flag
```
Comment (Tkael) [inline schemas/commodity-README.md:86]: If your app only refreshes CAPI in response to active journal events and does not refresh CAPI `/market` data while the player is not actively playing the game then you may use the value from the Journal `LoadGame` event.

File: schemas/fssdiscoveryscan-README.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+#### Remove _Localised key/values
+All keys whose name ends with `_Localised`, i.e. the `Name_Localised`
+key/values in Items.
+
+#### Other Elisions
+You MUST remove the following key/value pairs from the data:
+
+  - `Progress` key/value pair.
+
+#### Item Category
```
Comment (Tkael) [inline schemas/fssdiscoveryscan-README.md:28]: I do not have any record of `FSSDiscoveryScan` events with `Category` or `Category_Localized` properties.

File: schemas/fssdiscoveryscan-README.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+## Senders
+The primary data source for this schema is the ED Journal event 
+`FSSDiscoveryScan`.
+
+### Key Renames
+Many of the key names have a different case defined in this schema, make 
+sure you are renaming them as appropriate.
+
+### Elisions
+#### Remove _Localised key/values
```
Comment (Tkael) [inline schemas/fssdiscoveryscan-README.md:19]: I do not have any record of `FSSDiscoveryScan` events with `Localised` properties.

File: schemas/blackmarket-README.md (hunk 1 above)
Comment (Athanasius) [reply schemas/blackmarket-README.md:13]: That'll be me failing to replace copy/pasted parts, will fix later.

File: schemas/blackmarket-README.md (hunk 2 above)
Comment (Athanasius) [reply schemas/blackmarket-README.md:38]: True, but really I think we're aiming at removing this schema entirely.

I *think* I based this elision on what the removed EDMC code did back in 2017, as it was the only reference I could find.

File: schemas/commodity-README.md (hunk 4 above)
Comment (Athanasius) [reply schemas/commodity-README.md:78]: But good luck enforcing that, particularly if your app explicitly allows for running it on another machine that only has access to the journals via network share, i.e. you have no way to be sure the game is actually running.

File: schemas/fssdiscoveryscan-README.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+`Category` key/value pair from each Item.
+
+### Augmentations
+#### horizons flag
+Use the value from the `LoadGame` event.
+#### odyssey flag
+Use the value from the `LoadGame` event.
+
+#### StarPos
+You MUST add a `StarPos` array containing the system co-ordinates from the
```
Comment (clonedArtie) [inline schemas/fssdiscoveryscan-README.md:40]: Be aware that the coordinates may not be always available. For example, sometimes the `Location` event may be written AFTER `FSSSignalDiscovered`. To make implementation easier, let's change the keyword `MUST` to `SHOULD`?

File: schemas/fssdiscoveryscan-README.md (hunk 7 above)
Comment (clonedArtie) [reply schemas/fssdiscoveryscan-README.md:19]: There may be stuff like `SignalName_Localised` property present.

File: schemas/blackmarket-README.md (hunk 2 above)
Comment (Athanasius) [reply schemas/blackmarket-README.md:38]: Ah, the EDMC code doesn't *include* it, which I took to imply it should be removed, but that's not true.  Removed from that section of the docs.

File: schemas/commodity-README.md (hunk 3 above)
Comment (Athanasius) [reply schemas/commodity-README.md:51]: I was leaving this as implied, as it's in the schema.  I can add something based on your comment though.

**Pull request merged**